python/pyspark/sql/readwriter.py

```diff
@@ -193,8 +193,9 @@ def json(self, path, schema=None, primitivesAsString=None, prefersDecimal=None,
 
                 *  ``PERMISSIVE`` : sets other fields to ``null`` when it meets a corrupted \
                   record and puts the malformed string into a new field configured by \
-                 ``columnNameOfCorruptRecord``. When a schema is set by user, it sets \
-                 ``null`` for extra fields.
+                 ``columnNameOfCorruptRecord``. An user-defined schema can include \
+                 a string type field named ``columnNameOfCorruptRecord`` for corrupt records. \
+                 When a schema is set by user, it sets ``null`` for extra fields.
```

Comment:
No. The two mode does not set null. In `failFast` mode, it fails a job. In `dropMalformed` mode, it drops the malformed lines whose length is shorter or longer.